test/unit/overwrite.spec.js

```diff
@@ -0,0 +1,84 @@
+import postscribe from '../../dist/postscribe';
+
+describe('document.write overwriting.', function() {
+
+  const readNativeDocumentMethodString = method => {
+    // Cache because this takes a long time.
+    let cache = readNativeDocumentMethodString.cache;
+    if (!cache) {
+      cache = readNativeDocumentMethodString.cache = {};
+    }
+    let result = cache[method];
+    if (result) {
+      return result;
+    }
+    const iframe = document.createElement('iframe');
+    document.body.appendChild(iframe);
+    result = cache[method] = String(iframe.contentDocument[method]);
+    iframe.parentNode.removeChild(iframe);
+    return result;
+  };
+
+  let isNative = method => {
+    return String(document[method]) === readNativeDocumentMethodString(method);
+  };
+
+  // Must be async to avoid polluting doc.write for the next tests.
+  it('overrides document.write for normal scripts.', done => {
+    expect(isNative('write')).to.be.ok();
+    postscribe(document.body, '<script src="remote/describe-write.js"></script>', {
+      releaseAsync: true,
+      done
+    });
+    expect(isNative('write')).not.to.be.ok();
+  });
+
+  it('does not override document.write for async scripts.', done => {
+    expect(isNative('write')).to.be.ok();
+    postscribe(document.body, '<script async src="remote/describe-write.js"></script>', {
+      releaseAsync: true,
+      done
+    });
+    expect(isNative('write')).not.to.be.ok();
+  });
+
+  it('afterAsync fires when async ignored.', done => {
+    postscribe(document.body, '<script async src="remote/describe-write.js"></script>', {
+      releaseAsync: false,
+      afterAsync: () => {
+        expect(1);
```

Comment:
Why do you need `expect(1)` here?  Wouldn't the absence of calling `done` fail the test?